src/server/common.h

```diff
@@ -156,6 +166,8 @@ bool ParseDouble(std::string_view src, double* value);
 const char* ObjTypeName(int type);
 
 const char* RdbTypeName(unsigned type);
+bool AbslParseFlag(std::string_view in, MaxMemoryFlag* flag, std::string* err);
+std::string AbslUnparseFlag(const dfly::MaxMemoryFlag& flag);
```

Comment:
Oh, I see that you're defining in the flag in another `.cc` file. I'd personally prefer to have the parsing logic alongside the flag, but your call.

Reply:
the reason for this is that I must declare the flag in dfly_main so I can not just define everything in main_service.cc

Reply:
Why can't you define it along with the parsers in `main_service.cc` and `DECLARE_` it in `dfly_main.cc`?

Reply:
i tried it and it did not work with the ADL. ABSL could not find those parse/unparse functions. You can checkout the branch and try yourself.

Reply:
My proposal:

Define all of the below in `main_service.cc`:
* MaxMemoryFlag
* Parse/Unparse
* FLAG itself

Then expose a simple getter / setter in `main_service.h`

Advantages:
* The fact that max memory is tracked by a flag is now an implementation detail
* Exposed header is cleaner (doesn't pass through `absl::[SG]etFlag` and doesn't expose `.value`)
* We can easily move to having this value saved elsewhere, like thread local if need be

I don't think it matters greatly, but since this is the first flag of all the registry, it's something to think about.

This will still support setting flags in via a "human readable" string, however it will bypass the `SetMaxMemory()` function above which is not ideal.

Anyway, as I said, just a proposal. Do what you think with it.
https://github.com/dragonflydb/dragonfly/compare/ConfigPrProposal?expand=1

Reply:
BTW perhaps it's only for this specific flag that we'll need to provide an explicit setter, I imagine that most flags are not changed from the code.